c10/core/TensorImpl.cpp

```diff
@@ -51,6 +51,13 @@ const at::Tensor& TensorImpl::_fw_grad(uint64_t level, const at::Tensor& self) c
   return autograd_meta_->fw_grad(level, self);
 }
 
+bool TensorImpl::_has_fw_grad() const {
+  // One reason for autograd_meta_ to be undefined is autograd isnt enabled.
+  // However in functions like _fw_grad the fallback code path also requires
+  // autograd to be enabled. This function can serve as a guard then.
+  return autograd_meta_ != nullptr;
```

Comment:
This will return True even if there is no actual fw grad.
If you want a function that just checks for the existence of autograd meta, you can just use `autograd_meta()` and check if it is null.